google-http-client/src/test/java/com/google/api/client/util/DateTimeTest.java

```diff
@@ -225,8 +220,18 @@ public void testParseRfc3339ToSecondsAndNanos() {
     assertParsedRfc3339(
         "2018-03-01T10:11:12.1000Z", SecondsAndNanos.ofSecondsAndNanos(1519899072L, 100000000));
   }
+  
+  public void testEpoch() {
+    assertParsedRfc3339(
+        "1970-01-01T00:00:00.000Z", SecondsAndNanos.ofSecondsAndNanos(0, 0));
+  }
+
+  public void testOneSecondBeforeEpoch() {
+    assertParsedRfc3339(
+        "1969-12-31T23:59:59.000Z", SecondsAndNanos.ofSecondsAndNanos(-1, 0));
+  }
```

Comment:
Would you add the following test too?

```
public void testGregorianCalendarStartDate() {
    assertParsedRfc3339(
        "1582-10-15T01:23:45.123Z", SecondsAndNanos.ofSecondsAndNanos(-12219287774L, -877000000));
}
```
(not 100% sure about the value of nanosecond part to represent `-12219287774877` milliseconds)

As per Beam's PubsubClientTest, date 1582-10-15 is an important date when Gregorian Calendar was adopted.
https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubClientTest.java#L144

Reply:
I can't verify that this value is correct, so I don't want to add it here.

Reply:
I will try to add this in a separate PR then.